### ISSUE TITLE
Rules should not be self exclusive 

### ISSUE THREAD

**Issue by Maxouwell** [troyhunt/password-purgatory-api#45]
Right now the "unique character" rules is incompatible with some others rules .

It defeats the objective of wasting the maximum of time of the user, since they would abandon following this rule.

I think this rule should be disabled or upgraded to something compatible with the rest (2 characters ?)

**Comment by MichaelNMaggs** [comment]
Agreed. It's a more general problem, though: the palindrome rule is incompatible with cat and dog

**Comment by zgael** [comment]
I agree as well, opened an issue regarding the "unique character" rule and the "must contain repeating characters" rule last week (on the wrong project, sorry 😢  ) : https://github.com/troyhunt/password-purgatory/issues/7

Now, a similar problem is actually still present between the  "Password must contain only unique characters" rule and the bobcat one, that only allows : 
bobcat => double B
Lynx rufus => double U
L. rufus => double U

Maybe it was the other rule (the "only unique characters") that shoud be removed, as it can easily become incompatible with other future rules ?

I recognize that it's a fun rule, kinda infuriating (especially when looking for Simpsons/Griffins family that do not share the same letters) - but it seems a bit too limiting, and will hinder adding more complicated rules later.

**Comment by MichaelNMaggs** [comment]
You are quite right. As you suggest, it would be best to remove the "Password must contain only unique characters" rule, as it's very limiting for both existing and for future requirements

**Comment by troyhunt** [comment]
I *mostly* agree with the feedback here, but I also think a blanket "rules cannot be self-exclusive" robs us of some good candidates. I'm more inclined to say they cannot be self-exclusive *when the infuration level is "ridiculous"*, or similar. To that effect, I've just pushed a change for the unique characters requirement to be "ridiculous" and I encourage any further PRs to the same effect.

**Comment by troyhunt** [comment]
Oh there definitely shouldn't ever be a successful password, the goal is purely to extend the period of time people waste *trying* to find a successful one 😈

**Comment by Maxouwell** [comment]
Well then it renders the "final" message "Password already in use" useless and not reachable, which is the shame as it is the funniest and also allow the ride to continue for really stubborn people.
Right now, the incompatible rules just prove to the password-enterer that it's a game that cannot go further (which is also a shame).

**Comment by troyhunt** [comment]
Let's just consider it a work in progress that'll probably evolve more over time 🙂

**Comment by AyeBraine** [comment]
That's a shame, I hoped to finish the game. Palindrome killed me at step 10 or so

**Comment by phillmac** [comment]
I can't get past this one because of the must be 12-16 characters in length rule. Anybody have any ideas? catbobcat‮2Lisafalldog
